download_weather_incremental.py

```
@@ -0,0 +1,366 @@
+        logger.info(f"Total time: {total_time/60:.1f} minutes")
+        logger.info(f"Average speed: {avg_speed:.1f} measurements/second")
+        logger.info(f"Data saved to: {output_file}")
+        logger.info("=" * 80)
+        
+        # Analyze dataset if requested
+        if analyze and total_measurements > 0:
+            logger.info("Analyzing dataset...")
+            try:
+                analysis_results = analyze_dataset(output_file)
+                logger.info("\nDataset Analysis:")
+                logger.info(f"  Total rows: {analysis_results['total_rows']:,}")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)